Content.Shared/Blocking/SharedBlockingSystem.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Audio;
+using Content.Shared.Damage;
+using Content.Shared.Damage.Prototypes;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Item;
+using Content.Shared.Physics;
+using Content.Shared.Popups;
+using Content.Shared.Toggleable;
+using Robust.Shared.Audio;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Collision.Shapes;
+using Robust.Shared.Physics.Dynamics;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Blocking;
+
+public sealed class SharedBlockingSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly FixtureSystem _fixtureSystem = default!;
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SharedBlockingComponent, GettingPickedUpAttemptEvent>(OnPickupAttempt);
+        SubscribeLocalEvent<SharedBlockingComponent, DroppedEvent>(OnDrop);
+
+        SubscribeLocalEvent<SharedBlockingComponent, GetItemActionsEvent>(OnGetActions);
+        SubscribeLocalEvent<SharedBlockingComponent, ToggleActionEvent>(OnToggleAction);
+
+        SubscribeLocalEvent<SharedBlockingUserComponent, DamageModifyEvent>(OnUserDamageModified);
+
+        SubscribeLocalEvent<SharedBlockingComponent, ComponentShutdown>(OnShutdown);
+    }
+
+    private void OnPickupAttempt(EntityUid uid, SharedBlockingComponent component, GettingPickedUpAttemptEvent args)
+    {
+        component.User = args.User;
+
+        //To make sure that this bodytype doesn't get set as anything but the original
+        if (TryComp(args.User, out PhysicsComponent? physicsComponent) && physicsComponent.BodyType != BodyType.Static
+                                                                       && !TryComp(args.User, out SharedBlockingUserComponent? blockingUserComponent))
```

Comment:
The generics versions should be used where possible.

Reply:
Can you explain or give me an example of what a generic version is?

Reply:
I can't use that here.
![image](https://user-images.githubusercontent.com/54602815/175325465-0601fc96-510d-4a91-ae96-6bc681570d92.png)

Reply:
TryComp<T>(uid, out var comp)